cfme/networks/provider/nuage.py

```diff
@@ -0,0 +1,64 @@
+import attr
+from widgetastic.widget import View
+from widgetastic_patternfly import Tab, BootstrapSelect, Input
+
+from cfme.common.provider import DefaultEndpoint, DefaultEndpointForm
+from cfme.common.provider_views import BeforeFillMixin
+from cfme.networks.security_group import SecurityGroupCollection
+from cfme.utils import version
+from . import NetworkProvider
+
+
+class NuageEndpoint(DefaultEndpoint):
+    @property
+    def view_value_mapping(self):
+        return {'security_protocol': self.security_protocol,
+                'hostname': self.hostname,
+                'api_port': getattr(self, 'api_port', 5000)
+                }
+
+
+class NuageEndpointForm(View):
+    @View.nested
+    class default(Tab, DefaultEndpointForm, BeforeFillMixin):  # NOQA
+        TAB_NAME = 'Default'
+        security_protocol = BootstrapSelect('default_security_protocol')
+        api_port = Input('default_api_port')
+
```

Comment:
**Question**
`hostname` in `NuageEndpoint.view_value_mapping`, but no widget here?